File: src/main/java/com/assets/board/service/TaxService.java
package com.assets.board.service;

import com.assets.board.dto.TotalTaxReportDto;
import org.springframework.web.multipart.MultipartFile;

public interface TaxService {

    TotalTaxReportDto calculateDividendTax(short year, MultipartFile ibReportFile, boolean isMilitary);

    void generateUaTaxDeclarationXml(short year) throws Exception;
}

File: src/main/java/com/assets/board/service/TaxService.kt
package com.assets.board.service

import com.assets.board.dto.TotalTaxReportDto
import org.springframework.web.multipart.MultipartFile

interface TaxService {
    fun calculateDividendTax(year: Short, ibReportFile: MultipartFile, isMilitary: Boolean): TotalTaxReportDto

    @Throws(Exception::class)
    fun generateUaTaxDeclarationXml(year: Short)
}
